framework/include/limiters/Limiter.h

```diff
@@ -61,23 +63,280 @@ class Limiter
+   * gradient vectors, direction vector, maximum and minimum allowable values, and
+   * geometric information from the face and cell centroids. It must be overridden
+   * by any derived class implementing a specific limiting strategy.
+   *
+   * @tparam T The data type of the scalar values and the return type.
+   * @param phi_upwind The scalar value at the upwind location.
+   * @param phi_downwind The scalar value at the downwind location.
+   * @param grad_phi_upwind Pointer to the gradient vector at the upwind location.
+   * @param grad_phi_downwind Pointer to the gradient vector at the downwind location.
+   * @param dCD A constant direction vector representing the direction of the cell face.
+   * @param max_value The maximum allowable value.
+   * @param min_value The minimum allowable value.
+   * @param fi FaceInfo object containing geometric details such as face centroid and cell
+   * centroids.
+   * @param fi_elem_is_upwind Boolean indicating if the face info element is upwind.
+   * @return The computed flux limiting ratio.
+   *
+   * This pure virtual function is intended to be defined in derived classes, which will implement
+   * the specific limiting algorithm. Derived classes will provide the actual logic for computing
+   * the flux limiting ratio, ensuring that the result adheres to the constraints and properties
+   * required by the specific limiting method.
+   *
+   * Here is an example of how a derived class might implement this method:
+   *
+   * @example
+   * @code
+   * class MyLimiter : public Limiter<Real>
+   * {
+   * public:
+   *     Real limit(const Real & phi_upwind,
+   *                const Real & phi_downwind,
+   *                const VectorValue<Real> * grad_phi_upwind,
+   *                const VectorValue<Real> * grad_phi_downwind,
+   *                const RealVectorValue & dCD,
+   *                const Real & max_value,
+   *                const Real & min_value,
+   *                const FaceInfo * fi,
+   *                const bool & fi_elem_is_upwind) const override
+   *     {
+   *         // Implementation of the specific limiting algorithm.
+   *         Real ratio = ... // Compute the ratio.
+   *         return ratio; // Return the computed ratio.
+   *     }
+   * };
+   * @endcode
    */
   virtual T limit(const T & phi_upwind,
                   const T & phi_downwind,
                   const VectorValue<T> * grad_phi_upwind,
-                  const RealVectorValue & dCD) const = 0;
+                  const VectorValue<T> * grad_phi_downwind,
+                  const RealVectorValue & dCD,
+                  const T & max_value,
+                  const T & min_value,
+                  const FaceInfo * fi,
+                  const bool & fi_elem_is_upwind) const = 0;
   virtual bool constant() const = 0;
   virtual InterpMethod interpMethod() const = 0;
+
+  /**
+   * @brief Functor for applying tsimplified slope limiting.
```

Comment:
```suggestion
   * @brief Functor for applying simplified slope limiting.
```

Reply:
It is hard to get rid of this diff since I has to move `FaceArg` down the header as it is now taking a `Moose::StateArg` that defines the state in which the limiters are computed. Is there any way that I can force git not to take a diff?